crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,786 @@
+                &config,
+                &counterparty_chain_id,
+                rt.clone(),
+                |chain_config| {
+                    if let Some(key_name) = key_name {
+                        chain_config.key_name = key_name.to_string();
+                    }
+                },
+            )?;
+
+            chains.insert(counterparty_chain_id.clone(), chain_handle);
+        };
+
+        let chain_handle = chains.get(chain.id()).unwrap();
+        let counterparty_chain_handle = chains.get(&counterparty_chain_id).unwrap();
+
+        let counterparty_client = ForeignClient::restore(
+            counterparty_client_id.clone(),
+            counterparty_chain_handle.clone(),
+            chain_handle.clone(),
+        );
+
+        let result = submit_light_client_attack_evidence(
+            &evidence,
+            chain,
+            counterparty_client,
+            counterparty_client_id,
+            counterparty_chain_handle,
+            misbehaviour,
+        );
+
+        if let Err(error) = result {
+            error!("{error}");
+        }
+    }
+
+    Ok(())
+}
+
+fn submit_light_client_attack_evidence(
+    evidence: &LightClientAttackEvidence,
+    chain: &CosmosSdkChain,
+    counterparty_client: ForeignClient<BaseChainHandle, BaseChainHandle>,
+    counterparty_client_id: ClientId,
+    counterparty: &BaseChainHandle,
+    misbehaviour: TendermintMisbehaviour,
+) -> Result<(), eyre::Error> {
+    info!(
+        "building light client attack evidence for client `{}` on chain `{}`",
+        counterparty_client_id,
+        counterparty.id(),
+    );
+
+    let counterparty_is_provider = is_counterparty_provider(chain, counterparty);
+    let counterparty_client_is_frozen = counterparty_client.is_frozen();
+
+    if !counterparty_is_provider && counterparty_client_is_frozen {
+        warn!(
+            "cannot submit light client attack evidence to client `{}` on chain `{}`",
+            counterparty_client_id,
+            counterparty.id()
+        );
+        warn!("reason: client is frozen and chain is not a CCV provider chain");
+
+        return Ok(());
+    }
+
+    let signer = counterparty.get_signer()?;
+    let common_height = Height::from_tm(evidence.common_height, chain.id());
+
+    let provider_has_common_consensus_state = counterparty_is_provider
+        && has_consensus_state(counterparty, &counterparty_client_id, common_height);
+
+    if counterparty_is_provider
+        && counterparty_client_is_frozen
+        && !provider_has_common_consensus_state
+    {
+        warn!(
+            "cannot submit light client attack evidence to client `{}` on provider chain `{}`",
+            counterparty_client_id,
+            counterparty.id()
+        );
+        warn!("reason: client is frozen and does not have a consensus state at height {common_height}");
+
+        return Ok(());
+    }
+
+    let mut msgs = match counterparty_client.wait_and_build_update_client(common_height) {
```

Comment:
Should we update client if `provider_has_common_consensus_state`?

Reply:
I had missed that, good catch! No need indeed, since the provider already has that consensus state.

Reply:
https://github.com/informalsystems/hermes/pull/3456/commits/3e8b37f58a47dca3c6da72dcc597cf7833bfa192

Reply:
Can we do the same for sovereign chains? i.e. rename `provider_has_common_consensus_state` to `counterparty_has_common_consensus_state` and drop the `counterparty_is_provider` check from its definition. Then, if `true`, skip client update for both provider and sovereign chain as well.